### PR TITLE
upgrade scalaz and use latest features

### PR DESCRIPTION
**Pull request by fommil**

### CODE REVIEW

File: modules/scalaz/shared/src/test/scala-2.12/eu/timepit/refined/scalaz/RefTypeSpecScalazMonadError.scala [hunk 1]
```diff
@@ -30,13 +30,8 @@ object Decoder {
...
       def to[A](fa: Decoder[A]) = ReaderT[Out, String, A](fa.decode)
     }
-  // will be cleaner after https://github.com/scalaz/scalaz/pull/1661
-  private[this] val ME = MonadError[MT, String]
   implicit def monad: MonadError[Decoder, String] =
-    new IsomorphismMonadError[Decoder, MT, String] {
-      override implicit val G: MonadError[MT, String] = ME
-      override val iso: Decoder <~> MT = isoReaderT
-    }
+    MonadError.fromIso(isoReaderT)
```
Comment (fommil) [inline modules/scalaz/shared/src/test/scala-2.12/eu/timepit/refined/scalaz/RefTypeSpecScalazMonadError.scala]: `.fromIso` was added specifically to help with this usecase.

**Comment by fommil** [comment]
it looks to me like codecov doesn't know how to count :smirk_cat:

**Comment by fthomas** [comment]
LGTM, thanks @fommil!

**Comment by fommil** [comment]
FYI @fthomas we're investigating how to make the `Iso` definition simpler in https://github.com/scalaz/scalaz/pull/1696 ... we seem to have hit a compiler bug.

**Pull request merged**